Sources/System/FileOperations.swift

```diff
@@ -370,4 +370,52 @@ extension FileDescriptor {
   public func dup2() throws -> FileDescriptor {
     fatalError("Not implemented")
   }
+  
+  /// A pair of `FileDescriptor` values represening a pipe.
+  ///
+  /// You are responsible for managing the lifetime and validity
+  /// of the `read` and `write` `FileDescriptor` values,
+  /// which can be closed independently or via helper methods on `Pipe`, but not both.
+  public struct Pipe {
+    /// The file descriptor for reading from this pipe
+    public let fileDescriptorForReading: FileDescriptor
```

Comment:
But is it "input" or "output"? The end for reading gives you the content of `write` calls, so it's the output of a pipe, or am I mixing this up? Naming is very important here...

Reply:
It may be very Objective-C of me, but this is why I initially went with the `fileDescriptorForReading` (this mirrors prior art in `Foundation.Pipe`). I tend to prefer more verbose names, except in cases where there is a sufficiently descriptive term-of-art. I'm by no means an expert in POSIX, so I'll defer to you.

Reply:
I reverted this back to use the more verbose names, which I prefer more.

Reply:
The terms `input` and `output` in my suggestion reflect the naming convention as defined by `FileDescriptor.standardInput` and `.standardOutput` -- an "input filedescriptor" is one that you can read from; an "output fd" is one you can write to.

```swift
let (input, output) = FileDescriptor.pipe()
defer { 
  try! input.close() // Closing a valid pipe never errors out.
  try! output.close()
}
try launchChildProcess(input: .standardInput, output: output)
try launchChildProcess(input: input, output: .standardOutput)
```